src/main/scala/org/apache/spark/sql/catalyst/optimizer/RangerSparkAuthorizerExtension.scala

```diff
@@ -75,6 +79,30 @@ case class RangerSparkAuthorizerExtension(spark: SparkSession) extends Rule[Logi
     }
   }
 
+  private def authorizeSetCommandExcludedRules(exclusions: String) = {
+      if (containsRangerExtension(exclusions)) {
+        val ace = new SparkAccessControlException("Ranger extensions are not allowed to be excluded")
+        LOG.error(
+          s"""
+             |+===============================+
+             ||Spark SQL Authorization Failure|
+             ||-------------------------------|
+             ||${ace.getMessage}
+             ||-------------------------------|
+             ||Spark SQL Authorization Failure|
+             |+===============================+
+             """.stripMargin)
+        throw ace
+      }
+  }
+
+  private def containsRangerExtension(excluded: String) = {
+      excluded.contains(ru.typeTag[RangerSparkAuthorizerExtension].tpe.toString) ||
```

Comment:
Is it simpler if just `classOf[RangerSparkAuthorizerExtension].getName`,  or just check the package here like `classOf[RangerSparkAuthorizerExtension].getPackage.getName` in case we add more security extensions here.

Reply:
I considered it, but i found that this classes are on this package `org.apache.spark.sql.catalyst.optimizer`. This would prevent that some legitimate extensions will be excluded, isn't it?

Do you know why aren't those extensions on a non-spark namespace? Should I move them?

Reply:
This is because these classes might visit some private members in spark private packages. then, classOf[RangerSparkAuthorizerExtension].getName might be fair enough.